File: cz_app/lib/widget/app/models/employee.dart
class Employee {
  int id;
  int departmentId;
  String name;
  String email;
  int? referralCount;

  String? role;
  Employee(
      {required this.id,
      required this.departmentId,
      required this.name,
      required this.email,
      this.role,
      this.referralCount});

  factory Employee.fromJson(Map<String, dynamic> json) {
    return Employee(
        id: json['employee']['id'],
        departmentId: json['employee']['departmentId'],
        name: json['employee']['employeeName'],
        email: json['employee']['employeeEmail'],
        role: json['employee']['role'] as String?,
        referralCount: json['referralCount']);
  }
}

File: backend/Models/Employee.cs
namespace CZConnect.Models;

public class Employee
{
    public long Id { get; set; }
    public string EmployeeEmail { get; set; } = string.Empty;
    public long? DepartmentId { get;set; } 
    public Department? Department;
    public string EmployeeName { get; set; } = string.Empty;
    public EmployeeRole Role { get; set; }
    public string PasswordHash {get;set;} = string.Empty;
    
}
